x/mock/simulation/random_simulate_blocks.go

```diff
@@ -76,6 +78,16 @@ func SimulateFromSeed(
 	header := abci.Header{Height: 0, Time: timestamp}
 	opCount := 0
 
+	// Setup code to catch SIGTERM's
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		fmt.Printf("Exiting early due to SIGTERM, on block %d, operation %d\n", header.Height, opCount)
+		DisplayEvents(events)
+		os.Exit(1)
```

Comment:
It is quite conventional in the UNIX world to return 128 + `signal` code upon exit due to a signal, so I'd suggest to call `os.Exit(128 + int(syscall.SIGTERM))` instead of `os.Exit(1)`

Reply:
Excellent idea, let's do that.